File: Python_Scripting_for_Computational_Science_Third_Edition/py/intro/datatrans1.py
#!/usr/bin/env python
import sys, math

try:
    infilename  = sys.argv[1]
    outfilename = sys.argv[2]
except:
    print "Usage:",sys.argv[0], "infile outfile"
    sys.exit(1)
    
ifile = open( infilename, 'r')  # open file for reading
ofile = open(outfilename, 'w')  # open file for writing

def myfunc(y):
    if y >= 0.0:
        return y**5*math.exp(-y)
    else:
        return 0.0

# read ifile line by line and write out transformed values:
for line in ifile:
    pair = line.split()
    x = float(pair[0]); y = float(pair[1])
    fy = myfunc(y)  # transform y value
    ofile.write('%g  %12.5e\n' % (x,fy))
ifile.close();  ofile.close()

File: Python_Scripting_for_Computational_Science_Third_Edition/efficiency/datatrans/C/datatrans1.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

double myfunc(double y)
{
  if (y >= 0.0) {
    return pow(y,5.0)*exp(-y);
  } else {
    return 0.0;
  }
}

int main (int argc, char* argv[])
{
  FILE *ifile;  /* input  file */
  FILE *ofile;  /* outout file */
  double x, y; 
  char *infilename;
  char *outfilename;
  int  n;
  int  ok;

  /* abort if there are too few command-line arguments */
  if (argc < 3) {
    printf("Usage: %s infile outfile\n", argv[0]);  exit(1);
  } else {
    infilename = argv[1]; outfilename = argv[2];
  }
  printf("%s: converting %s to %s\n",argv[0],infilename,outfilename);
  ifile = fopen( infilename, "r"); /* open for reading */
  ofile = fopen(outfilename, "w"); /* open for writing */

  ok = 1;  /* boolean variable for not end of file */
  while (ok) {
    n = fscanf(ifile, "%lf%lf", &x, &y); /* read x and y */
    if (n == 2) {
      /* successful read in fscanf: */
      /* printf("%g %12.5e\n", x, y); */
      y = myfunc(y);
      fprintf(ofile, "%g %12.5e\n", x, y);
    } else {
      /* no more numbers */ ok = 0;
    }
  }
  fclose(ifile);  fclose(ofile);
  return 0;
}





